OrbitGl/SamplingReport.cpp

```diff
@@ -55,84 +51,84 @@ void SamplingReport::UpdateReport() {
-    m_SelectedCallstackIndex = 0;
-    m_CallstackDataView->SetCallStack(nullptr);
+  selected_sorted_callstack_report_ = profiler_->GetSortedCallstacksFromAddress(
+      selected_address_, selected_thread_id_);
+  if (selected_sorted_callstack_report_->callstacks_count.empty()) {
+    selected_sorted_callstack_report_ = nullptr;
+    selected_callstack_index_ = 0;
+    callstack_data_view_->SetCallStack(nullptr);
   } else {
-    OnCallstackIndexChanged(m_SelectedCallstackIndex);
+    OnCallstackIndexChanged(selected_callstack_index_);
   }
 }
 
-void SamplingReport::OnSelectAddress(uint64_t a_Address, int32_t a_ThreadId) {
-  if (m_CallstackDataView) {
-    if (m_SelectedAddress != a_Address || m_SelectedTid != a_ThreadId) {
-      m_SelectedSortedCallstackReport =
-          m_Profiler->GetSortedCallstacksFromAddress(a_Address, a_ThreadId);
-      m_SelectedAddress = a_Address;
-      m_SelectedTid = a_ThreadId;
+void SamplingReport::OnSelectAddress(uint64_t address, ThreadID thread_id) {
+  if (callstack_data_view_) {
+    if (selected_address_ != address || selected_thread_id_ != thread_id) {
+      selected_sorted_callstack_report_ =
+          profiler_->GetSortedCallstacksFromAddress(address, thread_id);
+      selected_address_ = address;
+      selected_thread_id_ = thread_id;
       OnCallstackIndexChanged(0);
     }
   }
 
-  if (m_UiRefreshFunc) {
-    m_UiRefreshFunc();
+  if (ui_refresh_func_) {
+    ui_refresh_func_();
   }
 }
 
 void SamplingReport::IncrementCallstackIndex() {
   CHECK(HasCallstacks());
-  size_t maxIndex =
-      m_SelectedSortedCallstackReport->callstacks_count.size() - 1;
-  if (++m_SelectedCallstackIndex > maxIndex) {
-    m_SelectedCallstackIndex = 0;
+  size_t max_index =
+      selected_sorted_callstack_report_->callstacks_count.size() - 1;
+  if (++selected_callstack_index_ > max_index) {
+    selected_callstack_index_ = 0;
   }
 
-  OnCallstackIndexChanged(m_SelectedCallstackIndex);
+  OnCallstackIndexChanged(selected_callstack_index_);
 }
 
 void SamplingReport::DecrementCallstackIndex() {
   CHECK(HasCallstacks());
-  size_t maxIndex =
-      m_SelectedSortedCallstackReport->callstacks_count.size() - 1;
-  if (m_SelectedCallstackIndex == 0) {
-    m_SelectedCallstackIndex = maxIndex;
+  size_t max_index =
+      selected_sorted_callstack_report_->callstacks_count.size() - 1;
+  if (selected_callstack_index_ == 0) {
+    selected_callstack_index_ = max_index;
   } else {
-    --m_SelectedCallstackIndex;
+    --selected_callstack_index_;
   }
 
-  OnCallstackIndexChanged(m_SelectedCallstackIndex);
+  OnCallstackIndexChanged(selected_callstack_index_);
 }
 
-std::string SamplingReport::GetSelectedCallstackString() {
-  if (m_SelectedSortedCallstackReport) {
-    int num_occurrences = m_SelectedSortedCallstackReport
-                              ->callstacks_count[m_SelectedCallstackIndex]
+const std::string& SamplingReport::GetSelectedCallstackString() const {
```

Comment:
Not sure, if this is correct. Doesn't it create the returned string only on the callee's stack. Thus might be invalid on the caller? Maybe I am wrong, then forget about my comment.

Reply:
I reverted the change

Reply:
Thanks